blockstore/apps/bundles/storage.py

```diff
@@ -0,0 +1,165 @@
+
+    Context:
+
+        When powering Blockstore with the S3SBoto3torageBackend, asset URLs are
+        exposed to clients (and users) by generating signed S3 URLs.
+        The ideal time-to-live of these URLs is ~24-48 hours;
+        that is, long enough such that users may interact with content without
+        losing access to assets in the middle of their session,
+        but short enough as to avoid granting perpetual unauthenticated access
+        to copyrighted content.
+
+        However, servers deployed on AWS are often given *temporary*
+        credentials instead of permanent ones. If those credentials are used
+        to sign S3 URLs, then the S3 URLs will automatically expire when
+        then the server's temporary credentials  expire (even if the URL was just
+        signed!). Thus, temporary credentials, although preferable for security
+        reasons, cannot be depended upon to sign our "long-lasting" S3 URLs.
+
+        The compromise here is to allow Blockstore to have two separate
+        sets of credentials:
+        1. temporary credentials that allow full read/write/delete access to S3
+           (along with any other necessary AWS permissions), and
+        2. permanent credentials that *only* grant read access to S3.
+
+        The first set of credentials remain the default credentials for Blockstore,
+        while the second set of credentials can be specifically used to sign
+        S3 URLs. Hence, this class.
+
+    THIS CLASS IS SPECIFIC TO THE S3Boto3Storage BACKEND.
+    It will not work with FileSystemStorage or any other backend.
+    (If it seemed prudent, though, this class could be generalized to work
+     with other storage backends.)
+    """
+
+    class BackendNotAvailable(Exception):
+        """
+        Raised if `LongLivedSignedUrlStorage` is not configured for use.
+        """
+        def __str__(self):
+            return (
+                f"In order to instantiate this storage backend, "
+                "boto3 must be installed "
+                "and both BUNDLE_ASSET_URL_STORAGE_KEY "
+                "and BUNDLE_ASSET_URL_STORAGE_SECRET "
+                "must be configured in Django settings."
+            )
+
+    def __init__(self):
+        """
+        Construct a storage backend instance.
+
+        Raises `BackendNotAvailable` if we are not configured.
+        """
+        try:
+            S3Boto3Storage = get_storage_class("storages.backends.s3boto3.S3Boto3Storage")
+        except ImportError as import_error:
+            raise self.BackendNotAvailable from import_error
+        try:
+            key = settings.BUNDLE_ASSET_URL_STORAGE_KEY
+            secret = settings.BUNDLE_ASSET_URL_STORAGE_SECRET
+        except AttributeError as attr_error:
+            raise self.BackendNotAvailable from attr_error
+        if not (key and secret):
+            raise self.BackendNotAvailable
+        self.s3_backend = S3Boto3Storage(
```

Comment:
@kdmccormick I think this is not correct, it looks like the variables names that need to be passed for the credentials are `access_key` and `secret_key`, not `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`

Reply:
Thank you :+1: